### PR TITLE
feat(wildcard): Creates a new element which represents a wildcard.

### LINKED ISSUE #605: WildcardRef needed
Currently it is modelled by TypeParameterRef, that is highly counter-intuitive and leads to bugs, when somebody overrides Visitor.visitCtTypeParameterReference() and expects to see only type parameter references, not wildcards! (Yes, I actually trapped into this problem and had long debugging)

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #605

### CODE REVIEW

**Comment by monperrus** [comment]
can you add a test case?

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

File: src/main/java/spoon/reflect/reference/CtWildcardReference.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.reference;
+
+public interface CtWildcardReference extends CtTypeParameterReference {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtWildcardReference.java]: add javadoc?

**Comment by monperrus** [comment]
new concept => update the metamodel figures?
thanks.

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

File: src/main/java/spoon/reflect/reference/CtWildcardReference.java [hunk 2]
```diff
@@ -16,6 +16,9 @@
  */
 package spoon.reflect.reference;
 
+/**
+ * This interface defines a reference to a wildcard (<code>?</code>).
```
Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtWildcardReference.java]: Represents  a wildcard in generic type annotations, i.e. the "?" (e.g. the "?" in <code>Collection<?></code> or <code>Collection<? extends List></code>).

**Pull request merged**